### PR TITLE
fix: restored wccom-from query parameter

### PR DESCRIPTION
**Pull request by rjchow**
Related to #https://github.com/Automattic/woocommerce.com/pull/18470

This PR must not be merged before the above, as it will cause the NUX login flow to be broken without the correct wccom-from parameter handling

## Proposed Changes

* The handling of the `wccom-from` query parameter in the Woo OAuth flow has been broken for some time, as it seems that the selector was expecting it to be a top level query parameter. In actuality, the query parameter was being inserted within the `redirect_to` URL's query parameter
* This PR fixes the selector, and in the process, has the consequence of uncovering some branches that have been `false` ever since the `wccom-from` parameter was broken. As far as I can tell, these branches are obsolete and I simply deleted them. 

- ** It is possible that I'm mistaken, and that there are existing login flows that insert the `wccom-from` parameter at the top level. Please sound out if you are aware **

- While investigating the code I also noticed that the non-woo branded partner login page (https://github.com/Automattic/wp-calypso/pull/64362) seems to have been broken by URL encoding at some point, so I fixed it (cc: @lsl)
- I also tested with the new WC Core Profiler login, which supersedes the previous login (https://github.com/Automattic/wp-calypso/pull/35193). As a result of this, I deleted the code paths that relate to that design, as they were being activated due to `wccom-from` detection being fixed

## Testing Instructions

1. If you're able to run the [woocommerce-start-dev-env environment](https://github.com/woocommerce/woocommerce-start-dev-env), use that. Otherwise you'll have to settle for modifying the login page to calypso.localhost instead.
1. Check out https://github.com/Automattic/wp-calypso/pull/83340 and run that too
3. Check out this branch go to `woocommerce.test/start` 
4. Go through the NUX while logged out
5. When you click on the WooExpress CTA you should be brought to the login page on WordPress.com
6. If you not using the start-dev-env, change `wordpress.com` to `calypso.localhost:3000`
7. Observe that you see `All Woo Express stores are powered by WordPress.com!` instead of `All Woo stores are powered by WordPress.com!`

image

- Also observe that if you go straight to "Login" from the woocommerce.test home page, you should see the `All Woo stores are powered by WordPress.com!` text copy
- Please also test around the other login flows with this wp-calypso PR, as I'm not too confident if I broke any other login flow. As far as I can tell, the wccom-from parameter has been null for awhile for the oauth login flow, so enabling it broke some things. I'm not sure if the non-oauth2 flow is still in use anywhere.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/blocks/login/index.jsx [hunk 1]
```diff
@@ -387,26 +369,26 @@ class Login extends Component {
 					);
 				} else {
 					headerText = <h3>{ translate( "Let's get started" ) }</h3>;
+					const poweredByWpCom =
+						wccomFrom === 'nux'
+							? translate( 'All Woo Express stores are powered by WordPress.com!' )
+							: translate( 'All Woo stores are powered by WordPress.com!' );
```
Comment (ilyasfoo) [inline client/blocks/login/index.jsx:416]: I'd suggest to temporarily update this to `Log in to WooCommerce.com with your WordPress.com account to connect your store and manage your extensions` until we confirm a better copy to fix the misleading messaging.

**Review by ilyasfoo — DISMISSED** [review]
Copy changes tested well. I've tested these flows:

1. NUX login
2. WooCommerce.com login
3. Core extension connect in WooCommerce > Extensions > My subscriptions

However, I have a minor comment regarding the currently misleading messaging in non-NUX screens.

Also, it seems like the changes have affected styling in login screen. To reproduce, go through NUX flow and when prompted with signup screen, click on `Log in`:

Before:

image


After:

image

File: client/signup/steps/user/index.jsx [hunk 2]
```diff
@@ -195,14 +195,38 @@ export class UserStep extends Component {
...
-						? translate(
-								"You'll need an account to complete your purchase and manage your subscription"
-						  )
-						: translate(
-								"You'll need an account to connect your store and manage your extensions"
-						  );
+				switch ( wccomFrom ) {
+					case 'cart':
+						subHeaderText = translate(
+							"You'll need an account to complete your purchase and manage your subscription"
```
Comment (ilyasfoo) [inline client/signup/steps/user/index.jsx:204]: I'm unsure how to test this flow 🤔

File: client/signup/steps/user/index.jsx [hunk 3]
```diff
@@ -195,14 +195,38 @@ export class UserStep extends Component {
...
+									br: <br />,
+								},
+								comment:
+									'Link displayed on the Signup page to users having account to log in WooCommerce via WordPress.com',
+							}
+						);
+						break;
+					default:
+						subHeaderText = translate(
+							'All Woo stores are powered by WordPress.com.{{br/}}Please create an account to continue. Already registered? {{a}}Log in{{/a}}',
```
Comment (ilyasfoo) [inline client/signup/steps/user/index.jsx]: Also update this copy if you agree to temporarily resolve the misleading messaging

File: client/state/login/utils.js [hunk 4]
```diff
@@ -192,12 +192,16 @@ export function isPartnerSignupQuery( currentQuery ) {
...
+			decodeURIComponent( currentQuery.redirect_to )
+		);
 	}
 
 	// Handles user creation through /start/wpcc?oauth2_redirect=...
 	if ( typeof currentQuery?.oauth2_redirect === 'string' ) {
-		return /woocommerce\.com\/partner-signup/.test( currentQuery.oauth2_redirect );
+		return /woocommerce\.(?:com|test)\/partner-signup/.test(
+			decodeURIComponent( currentQuery.oauth2_redirect )
+		);
```
Comment (lsl) [inline client/state/login/utils.js:204]: lgtm https://woocommerce.com/partner-signup/ now redirects with encoded an redirect_to

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (andfinally) [reply client/blocks/login/index.jsx:416]: I agree, that wording seems better in this situation. I think this is what the UI used to say.

File: client/signup/steps/user/index.jsx (hunk 2 above)
Comment (rjchow) [reply client/signup/steps/user/index.jsx:204]: Me neither tbh, I was tempted to remove it because it's dead code, but I also wanted to leave a signpost here for future development

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (rjchow) [reply client/blocks/login/index.jsx:416]: A bit of a pickle here because the new layout shows the account selector if wpcom account is logged in, and there's the sign up link for the user if they don't have an existing wpcom account. 

It would look like:

image
image

which is a bit repetitive

**Comment by rjchow** [comment]
> Copy changes tested well. I've tested these flows:
> 
>     1. NUX login
> 
>     2. WooCommerce.com login
> 
>     3. Core extension connect in WooCommerce > Extensions > My subscriptions
> 
> 
> However, I have a minor comment regarding the currently misleading messaging in non-NUX screens.
> 
> Also, it seems like the changes have affected styling in login screen. To reproduce, go through NUX flow and when prompted with signup screen, click on `Log in`:
> 

Resolved but leaving me in a bit of a paranoid state of mind,  because it resulted from a stray wccomFrom clause that I missed. This thing is a bit of a fossilized spaghetti

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (andfinally) [reply client/blocks/login/index.jsx:416]: I don't suppose we can do like

```
Log in to WooCommerce.com with your WordPress.com account to connect your store and manage your extensions.
Don't have an account? Sign up.
```

? If it's not easy to change the second line, we can go with `All Woo stores are powered by WordPress.com!` for now, thanks.

**Review by andfinally — COMMENTED** [review]
I had trouble with the `woocommerce-start-dev-env` environment, so I just changed the hostname in the browser address bar when I got to the `wordpress.com` pages. I tested these, and they worked as expected:

* NUX login
* WooCommerce.com login

I also tested the "login from cart" flow. (I rebased the WCCOM `fix/wccom-from-nux` branch to fix an issue on the product page.) I'm not sure if it worked properly, or if the behaviour is expected when I try to log in to WPCOM in local Calypso.

- In incognito browser, go to [product page](https://woocommerce.test/products/woocommerce-box-office/) and add product to cart.
- Proceed to checkout.
- Click on "log in" link in text "Enter the email you want to receive the order confirmation and use to access your account later. If you have an account, please log in."
- See `https://wordpress.com/log-in?client_id=50019&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50019%26state%3Dda3c5ea4b5e101ef274d6d8ff1a12b4670ac0bd2db6e8a7c3a8ae97f512bcd5f%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.test%252Fwc-api%252Fwpcom-signin%253Fnext%253Dcheckout%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1`.
- Change the domain on this URL to `http://calypso.localhost:3000` in browser address bar.
- See "All Woo stores are powered by WordPress.com!" page.
- Enter WPCOM credentials.
- See 2FA page. Change the domain, so URL is `http://calypso.localhost:3000/log-in/authenticator?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50019%26state%3Dda3c5ea4b5e101ef274d6d8ff1a12b4670ac0bd2db6e8a7c3a8ae97f512bcd5f%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.test%252Fwc-api%252Fwpcom-signin%253Fnext%253Dcheckout%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1&client_id=50019`
- Enter 2FA code.
- See WPCOM login page `https://wordpress.com/log-in?client_id=50019&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50019%26state%3Dda3c5ea4b5e101ef274d6d8ff1a12b4670ac0bd2db6e8a7c3a8ae97f512bcd5f%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.test%252Fwc-api%252Fwpcom-signin%253Fnext%253Dcheckout%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1`.

File: client/signup/steps/user/index.jsx (hunk 2 above)
Comment (andfinally) [reply client/signup/steps/user/index.jsx:204]: I agree it doesn't seem to be used. When I click "Log in" from the cart page on production, `wccom-from` is empty, and on logging in I'm sent back to the cart page.

When I add `wccom-from%3Dcart` to the Calypso URL, I get this UI. (I suppose my logged-in state is overwriting the subheader text.) This is quite nice and there are probably more parts to it than this, so it could be OK to leave it for now, in case anyone comes round to this approach again. It seems we've accidentally lost a few good things from this flow over the years. cc @poligilad-auto for your interest.

image

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (andfinally) [reply client/blocks/login/index.jsx:416]: On consideration, I find the "All Woo stores are powered" text a bit weird. I've asked for clarification about it on p6JqRr-bbu.

**Comment by poligilad-auto** [comment]
> It seems we've accidentally lost a few good things from this flow over the years. cc @poligilad-auto for your interest.

Thanks @andfinally. While it's interesting, it's kind of confusion too, we have to do a proper audit and reviw of the login screens to see what makes sense. I'm pretty sure we aren't using that type of pattern anymore.

image

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (andfinally) [reply client/blocks/login/index.jsx:416]: The writing experts are now suggesting this for the text. This may not fit neatly with the existing logic – or maybe it will help simplify things.

### Woo Express signup and log in

> All Woo Express stores are powered by WordPress.com. 
> Please log in to continue. Don’t have an account? [Sign up](link).

### Everything else

> To connect to services, manage extensions, and more, please log in using your WordPress.com account. Don’t have an account? [Create one](link).

**Comment by andfinally** [comment]
Just wanted to mention that we will likely revisit this next year. We have three items in our plan for the Marketplace. These flows are yet to be designed. They are:

- Design a Marketplace entry point in the NUX
- Optimize the sign-up flow using best practices from the A8C ID atelier
- Improve the onboarding / sign-up flow to have a few different paths (Just create account, Woo Express, Download WooCommerce)

But we needn't wait for those to firm up. Happy to jump on a call @rjchow, if it would help to find a way to simplify the wording!

**Comment by rjchow** [comment]
> Just wanted to mention that we will likely revisit this next year. We have three items in our plan for the Marketplace. These flows are yet to be designed. They are:
> 
>     * Design a Marketplace entry point in the NUX
> 
>     * Optimize the sign-up flow using best practices from the A8C ID atelier
> 
>     * Improve the onboarding / sign-up flow to have a few different paths (Just create account, Woo Express, Download WooCommerce)
> 
> 
> But we needn't wait for those to firm up. Happy to jump on a call @rjchow, if it would help to find a way to simplify the wording!

Hey @andfinally sorry about the gap! I was on AFK and just got back, so this is one of the top items on my agenda. I'll get this sorted soon

File: client/blocks/login/index.jsx (hunk 1 above)
Comment (rjchow) [reply client/blocks/login/index.jsx:416]: I had a bit of an issue with a dangling word due to the width constraint so I took a bit of liberty with the wording, hope this is ok?

### Non-NUX flows:
###### Sign up
image

###### Log in
image


### NUX
###### Sign up
image

###### Log in
image

Comment (rjchow) [reply client/blocks/login/index.jsx:416]: ^ pinging @andfinally in case it got lost in your notifications 😬

**Review by chihsuan — APPROVED** [review]
Nice work! @rjchow 

I tweaked the two-factor screen styles, and I've tested these flows:

1. NUX
2. WooCommerce.com login
3. Core extension connect in WooCommerce > Extensions > My subscriptions
4. Checkout flow
5. Core profiler flow
6. Partner flow
7. Woo DNA flow

Confirmed everything looks good! 🚀
